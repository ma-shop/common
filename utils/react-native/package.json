{
  "name": "@ma-shop/react-native",
  "version": "0.0.20",
  "main": "dist/index.js",
  "scripts": {},
  "author": "Tyler Benton",
  "license": "MIT",
  "publishConfig": {
    "access": "public"
  },
  "gitHead": "3e213cdf455e3ac36b88a85e1f7c43d1936fe50d",
  "peerDependencies": {
    "@ma-shop/is": "*",
    "@ma-shop/utils": "*",
    "react": "*",
    "react-native": "*"
  },
  "devDependencies": {
    "@ma-shop/is": "^0.0.11",
    "@ma-shop/utils": "^0.0.18"
  },
  "dependencies": {
    "prop-types": "15.7.2"
  }
}
