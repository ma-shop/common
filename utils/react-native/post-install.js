const fs = require('fs').promises


async function run () {
  const basePath = `${process.cwd()}/node_modules/@ma-shop`
  const packagesPath = `${basePath}/utils/dist/packages.js`
  const file = `${await fs.readFile(packagesPath).catch(() => '')}`
  const packages = await fs.readdir(basePath).catch(() => [])

  if (file && packages.length) {
    const splitter = 'try {\n'
    const filteredPackages = file
      .split(splitter)
      .filter((str) => !str.includes('@ma-shop') || packages.find((pkg) => str.includes(pkg)))
      .join(splitter)

    if (filteredPackages) {
      await fs.writeFile(packagesPath, filteredPackages)
    }
  }
}

run()
