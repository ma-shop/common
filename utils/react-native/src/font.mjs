import { round, ui } from '@ma-shop/utils'

/// @name em
/// @description This function is used to mimic the `em` functionality of css
/// em's allow us to code things responsively in a very easy way
/// @arg {number} value - The em value you want to get
/// @returns {unit} - the px value based on the baseUnit size
export function em (value = 1, base) {
  return round(value * (base || ui.fontSize))
}

/// @name px
/// @description this will convert your pixel value to em's
/// @arg {number} value - the pixel to compute
/// @returns {unit} - the em value based on the base unit
export function px (value, base) {
  return round(value / (base || ui.fontSize))
}
