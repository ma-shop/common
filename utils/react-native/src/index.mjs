// !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
// !!! This file is generated do not modify it manually because it will be overwritten
// !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

export * from './get-device-locale'
export * from './interactions'
export * from './is'
export * from './navigation'
export * from './shapes'
