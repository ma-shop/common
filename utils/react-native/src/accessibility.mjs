import { PixelRatio } from 'react-native'
import { clamp } from '@ma-shop/utils'

/// @name accessibilityStates
/// @author Tyler Benton
/// @description This is used to handle the accessibilityStates prop in a better way
/// since dealing with an array of strings is a weird implementation
/// @arg {object} obj = The states that you want to set
/// @example
///
/// render () {
///   const { isDisabled } = this.state
///
///   return (
///     <View
///       accessible
///       accessibilityStates={accessibilityStates({ isDisabled })}
///     />
///   )
/// }
export function accessibilityStates (obj) {
  return Object.keys(obj)
    .map((key) => obj[key] && key.replace(/^is/, '').toLowerCase())
    .filter(Boolean)
}


/// @name fontScale
/// @author Tyler Benton
/// @description This allows you to fake the font scaling if it's needed.
/// This is typically used with Images but it can be used in other cases if needed.
/// @note When you change the font scale in the accessibility inspector it will not show changes
/// on the screen like the actual fonts show. To see changes take affect you will have to
/// reload the screen.
export function fontScale ({
  allowFontScaling = true,
  noFontScaling = false,
  maxFontSizeMultiplier = 2.25,
  minimumFontScale = 1,
} = {}) {
  if (!allowFontScaling || noFontScaling) return 1
  return clamp(PixelRatio.getFontScale(), minimumFontScale, maxFontSizeMultiplier)
}
