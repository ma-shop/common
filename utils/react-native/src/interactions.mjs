import { InteractionManager } from 'react-native'
import { delay } from '@ma-shop/utils'

/// @name interactions
/// @author Tyler Benton
/// @description This function is used to wait for other interactions to finish
/// before you run a specific interaction. It allows animtations to run smooth
/// because there's not as many animations running at a time
/// @arg {number} max [1000] - The max time you wait for interactions to finish
export function interactions (max = 1000) {
  return Promise.race([
    new Promise(InteractionManager.runAfterInteractions),
    delay(max),
  ])
}
