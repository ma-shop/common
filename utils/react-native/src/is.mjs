import { Platform } from 'react-native'
import { is } from '@ma-shop/is'


/// @name ios
/// @description This will check if the current device is on ios
/// @returns {boolean}
is.ios = () => Platform.OS === 'ios'

/// @name android
/// @description This will check if the current device is on android
/// @arg {array, string} - The types to test against
/// @returns {boolean}
is.android = () => Platform.OS === 'android'
