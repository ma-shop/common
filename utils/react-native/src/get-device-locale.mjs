import { NativeModules } from 'react-native'
import { is } from '@ma-shop/is'

/// @name getDeviceLocale
/// @description This function will get the current locale for where the device is.
/// @returns {string} - eg. en-US
// @todo Move this to the @ma-shop/react-native library
export function getDeviceLocale () {
  let deviceLocale

  if (is.ios()) {
    deviceLocale = NativeModules.SettingsManager.settings.AppleLocale
    if (!deviceLocale) {
      // iOS 13 workaround, take first of AppleLanguages array
      [ deviceLocale ] = NativeModules.SettingsManager.settings.AppleLanguages
    }
  } else {
    deviceLocale = NativeModules.I18nManager.localeIdentifier
  }

  // device locale isn't available; return en-us
  if (!deviceLocale) return 'en-US'

  const localePieces = deviceLocale.split('_')

  if (localePieces.length !== 2) return 'en-US'

  const language = localePieces[0].toLowerCase()
  const country = localePieces[1].toUpperCase()
  return `${language}-${country}`
}
