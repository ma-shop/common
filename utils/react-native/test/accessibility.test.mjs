import { PixelRatio } from 'react-native'

import { accessibilityStates, fontScale } from '../src/accessibility'


PixelRatio.getFontScale = jest.fn(() => 1)

describe('accessibility', () => {
  test('accessibilityStates', () => {
    const actual = accessibilityStates({ isDisabled: true, disabled: true, something: false })

    expect(actual).toMatchSnapshot()
  })

  describe('fontScale', () => {
    test('allowFontScaling and noFontScaling', () => {
      expect(fontScale({ allowFontScaling: false })).toBe(1)
      expect(fontScale({ noFontScaling: true })).toBe(1)
    })

    test('minimumFontScale', () => {
      PixelRatio.getFontScale.mockImplementationOnce(() => 0.5)

      expect(fontScale()).toBe(1)

      PixelRatio.getFontScale.mockImplementationOnce(() => 1.5)

      expect(fontScale()).toBe(1.5)
    })

    test('maxFontSizeMultiplier', () => {
      PixelRatio.getFontScale.mockImplementationOnce(() => 4)

      expect(fontScale()).toBe(2.25)

      PixelRatio.getFontScale.mockImplementationOnce(() => 1.5)

      expect(fontScale()).toBe(1.5)
    })
  })
})

