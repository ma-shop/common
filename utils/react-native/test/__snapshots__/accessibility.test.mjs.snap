// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accessibility accessibilityStates 1`] = `
Array [
  "disabled",
]
`;
